pallets/liquidity-mining/src/tests/update_global_farm.rs

```diff
@@ -182,3 +183,124 @@ fn update_global_farm_price_adjustment_should_fail_when_farm_is_already_terminat
+
+			set_block_number(100_000);
+
+			//Act
+			assert_ok!(LiquidityMining::update_global_farm(
+				GC_FARM,
+				planned_yielding_periods,
+				yield_per_period,
+				min_deposit,
+			));
+
+			//Assert that global farm is updated
+			assert_eq!(
+				LiquidityMining::global_farm(GC_FARM).unwrap(),
+				GlobalFarmData {
+					updated_at: 1_000,
+					planned_yielding_periods,
+					yield_per_period,
+					min_deposit,
+					accumulated_rpz: FixedU128::from_inner(491_000_000_000_000_000_000_u128),
+					pending_rewards: 343195125000000000000,
+					max_reward_per_period: 29655521325000,
+					..global_farm_0
+				},
+			);
+			frame_system::Pallet::<Test>::assert_has_event(mock::RuntimeEvent::LiquidityMining(
+				Event::GlobalFarmAccRPZUpdated {
+					global_farm_id: global_farm_0.id,
+					accumulated_rpz: FixedU128::from_inner(491_000_000_000_000_000_000_u128),
+					total_shares_z: global_farm_0.total_shares_z,
+				},
+			));
+
+			TransactionOutcome::Commit(DispatchResult::Ok(()))
+		});
+	});
+}
+
+#[test]
+fn update_global_farm_should_fail_with_invalid_deposit() {
+	predefined_test_ext_with_deposits().execute_with(|| {
+		let _ = with_transaction(|| {
+			//Arrange
+			let planned_yielding_periods: BlockNumber = 1_000_000_000_u64;
+			let yield_per_period = Perquintill::from_percent(20);
+
+			set_block_number(100_000);
+
+			//Act
+			assert_noop!(
+				LiquidityMining::update_global_farm(
+					GC_FARM,
+					planned_yielding_periods,
+					yield_per_period,
+					MIN_DEPOSIT - 1,
+				),
+				Error::<Test, Instance1>::InvalidMinDeposit
+			);
+
+			TransactionOutcome::Commit(DispatchResult::Ok(()))
+		});
+	});
+}
+
+#[test]
+fn update_global_farm_should_fail_when_planning_yield_period_is_too_small() {
+	predefined_test_ext_with_deposits().execute_with(|| {
+		let _ = with_transaction(|| {
+			//Arrange
+			let planned_yielding_periods: BlockNumber = MinPlannedYieldingPeriods::get() - 1;
+			let yield_per_period = Perquintill::from_percent(20);
+
+			set_block_number(100_000);
+
+			//Act
+			assert_noop!(
+				LiquidityMining::update_global_farm(GC_FARM, planned_yielding_periods, yield_per_period, MIN_DEPOSIT,),
+				Error::<Test, Instance1>::InvalidPlannedYieldingPeriods
+			);
+
+			TransactionOutcome::Commit(DispatchResult::Ok(()))
+		});
+	});
+}
+
+#[test]
+fn update_global_farm_should_fail_when_yield_period_is_zero() {
+	predefined_test_ext_with_deposits().execute_with(|| {
+		let _ = with_transaction(|| {
+			//Arrange
+			let planned_yielding_periods: BlockNumber = MinPlannedYieldingPeriods::get();
+			let zero_yield_per_period = Perquintill::from_percent(0);
+
+			set_block_number(100_000);
+
+			//Act
+			assert_noop!(
+				LiquidityMining::update_global_farm(
+					GC_FARM,
+					planned_yielding_periods,
+					zero_yield_per_period,
+					MIN_DEPOSIT,
+				),
+				Error::<Test, Instance1>::InvalidYieldPerPeriod
+			);
+
+			TransactionOutcome::Commit(DispatchResult::Ok(()))
+		});
+	});
+}
```

Comment:
I would like to also see test where user claimed correct amount after farms update, something like:
* run for few periods
* update global farm
* run for few periods
* claim assert for correct amount

Reply:
added, pls check it!